src/main/scripts/src/app/shared/perspectives/documents/post-it/document/post-it-document.component.html

```diff
@@ -1,33 +1,94 @@
+                       [textNoResults]="null"
+                       [ngModel]="attributePair.attribute"
+                       (ngModelChange)="updateAttribute(attributePair, $event)"
+                       (keydown)="onKeyDown($event)">
+        </ng2-completer>
+      </div>
+
+      <div id="AttributePair{{ index }}[1, {{ row }}]" tabindex="-1"
+           title="{{ attributePair.value }}"
+           (keydown)="onKeyDown($event)"
+           (click)="clickOnAttributePair(1, row)">
+
+        <input [ngModel]="attributePair.value"
+               (ngModelChange)="updateValue(attributePair, $event)"/>
+      </div>
+    </ng-template>
+
+    <ng-template #nonEditablePair>
+      <div class="attribute" title="{{ attributePair.attribute }}">
+        <input [disabled]="true" [ngModel]="attributePair.attribute">
+      </div>
+
+      <div title="{{ attributePair.value }}">
+        <input [disabled]="true" [ngModel]="attributePair.value">
+      </div>
+    </ng-template>
+
+  </li>
+
+  <li *ngIf="editable">
+    <div id="AttributePair{{ index }}[0, {{ attributePairs.length }}]" class="attribute" tabindex="-1"
+         (keydown)="onKeyDown($event)"
+         (click)="clickOnAttributePair(0, attributePairs.length)">
+
+      <ng2-completer placeholder=" + "
+                     [datasource]="suggestedAttributes"
+                     [minSearchLength]="0"
+                     [textNoResults]="null"
+                     [(ngModel)]="newAttributePair.attribute"
+                     (keydown)="onKeyDown($event)">
+      </ng2-completer>
+    </div>
+
+    <div id="AttributePair{{ index }}[1, {{ attributePairs.length }}]"
+         [attr.tabindex]="newAttributePair.attribute ? -1 : ''"
+         (keydown)="onKeyDown($event)"
+         (click)="newAttributePair.attribute && clickOnAttributePair(1, attributePairs.length)">
+
+      <input [disabled]="!newAttributePair.attribute"
+             (keydown)="$event.key.length === 1 && createAttributePair($event.key)"/>
+    </div>
+
+  </li>
+</ul>
 
 <!-- additional buttons -->
 <div class="edit">
   <span class="primary-edit">
-    <a class="fa fa-fw fa-search" aria-hidden="true" title="Details"></a>
+    <a class="fa fa-fw fa-search" title="Details"></a>
   </span>
 
   <span class="secondary-edit">
     <!--the comments remove added space between icons-->
-    <a class="fa fa-fw fa-sign-in" aria-hidden="true" title="Incoming Links"></a><!--
-    --><a class="fa fa-fw fa-sign-out" aria-hidden="true" title="Outgoing Links"></a><!--
-    --><a class="fa fa-fw fa-history" aria-hidden="true" title="History"></a><!--
-    --><a class="fa fa-fw fa-users" aria-hidden="true" title="User Rights"></a>
+    <a class="fa fa-fw fa-sign-in" title="Incoming Links"></a><!--
+    --><a class="fa fa-fw fa-sign-out" title="Outgoing Links"></a><!--
+    --><a class="fa fa-fw fa-history" title="History"></a><!--
+    --><a class="fa fa-fw fa-users" title="User Rights"></a>
```

Comment:
Why are there empty comments here?

Reply:
`<!--the comments remove added space between icons-->`
Right above it. :D
Without the comments I would have to put them all on same line in order to not have a space character inserted between the icons.
(Not sure what's the exact reason it happens, but it looks like html interprets all `\n` inside `<span>` as a space)